server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -141,18 +145,29 @@ public ClusterState execute(ClusterState localClusterState) throws Exception {
 
                     @Override
                     public void onFailure(String source, Exception e) {
-                        LOGGER.error(e);
+                        LOGGER.error("Tracking metadata failed for subscription {} {}", subscriptionName, e);
+                        countDown.onFailure(e);
+                    }
+
+                    @Override
+                    public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
```

Comment:
This method is called locally, so before all nodes acked to process the new cluster state.
Additionally it won't be called if the cluster state didn't change according to the method's comment and a short research by myself.
I think we must use either an `AckedClusterStateUpdateTask` or an impl of `AckedClusterStateTaskListener` here.